CONTRIBUTING.md

```diff
@@ -64,6 +64,16 @@ Next, to create a new change entry, in the root of the repository run: `changie
 
 Follow the prompts to create your change entry - remember this is what will show up in the changelog.  Changie registers the change in a .yaml file, and that file must be included in your pull request before we can release.
 
+### Auto Generating Types
+
+There is a code generator that helps maintain the various Enum types we use in graphql. It introspects the public Graphql Interface and keeps `enum.go` up to date. To use it:
+
+```
+go run gen.go
+```
+
+Once the GraphQL API is made public, and before adding any other types - try the above command to pull down any types that can be auto-generated.
```

Comment:
So the reason alot of Enums have been left out of the `enum.go` is because they are orphaned code.  There is alot still yet to be implemented and having the Enum's there would likely confuse an end user - "why is this enum here what is it used for?"

Now i'm not fully 100% against it I could be convinced otherwise - but i've been trying to keep the enum.go limited to only the things we actually end up using.  But I see how that can be hard with the generator.

So just relaying context - happy to allow in the unused enums if we feel strongly its a burden to maintain not committing unused ones.

Reply:
Ah ok, thank you for the context! I think it's confusing to have a generated file that we can only partially commit to. The file says "do not modify", but we still need to make manual modifications. Perhaps we could add the logic to omit certain fields in the generator that don't work (like the ContactType issue). 

I think it's fine having deprecated types in here though. I think the most important thing is the query interfaces, and the types just add validation on inputs/outputs so I would think most folks would just not interact with them.

Reply:
An example of where this confusion has already bit us is with the recent addition of `CheckServiceDependency` - It should have been `CheckTypeServiceDependency`. I will revert all of the other enum.go changes for now, but I _do_ think that one should go in imo. It's a simple rename and would prevent that change from happening every type we run `go generate`.